src/App.tsx

```diff
@@ -15,47 +15,103 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+export function getReorderedGoods(
+  goods: string[], sortType: SortType, isReversed: boolean,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((firstGood, secondGood) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return firstGood.localeCompare(secondGood);
+      case SortType.LENGTH:
+        return firstGood.length - secondGood.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+  const visibleGoods = getReorderedGoods(goodsFromServer, sortType, isReversed);
+
+  const sortAlphabetically = () => {
+    setSortType(SortType.ALPHABET);
+  };
+
+  const sortByLength = () => {
+    setSortType(SortType.LENGTH);
+  };
+
+  const Reversing = () => {
+    setIsReversed(!isReversed);
+  };
+
+  const Reserting = () => {
```

Comment:
```suggestion
  const reverse = () => {
    setIsReversed(!isReversed);
  };

  const reset = () => {
```
Use camelCase. Also, it would be great to name functions according to these rules:
![image](https://user-images.githubusercontent.com/96409563/232878254-302f89bc-3e27-4e77-9b4d-771cbabab42d.png)